### ISSUE TITLE
Invalid Name Inheritance 

### ISSUE THREAD

**Issue by serbboy23** [pylint-dev/pylint#8879]
### Bug description

I believe that if I disable pylint warnings for `invalid-name` where I declare a class level attribute, that should suppress errors where I use that attribute. 
```python

"""
This class shows a bug in pylint.
"""


class Base:
    """
    This is where the invalid-name field is defined.
    """

    # pylint: disable=invalid-name
    thisIsABadField: str
    # pylint: enable=invalid-name

    def __init__(self, greeting: str):
        print(greeting)

    def do_something(self):
        self.thisIsABadField = "hello"

    def do_another(self):
        self.thisIsABadField = "hello again"
```

### Configuration

```ini
[BASIC]
class-attribute-naming-style=snake_case
```

### Command used

```shell
pylint path/to/file.py
```

### Pylint output

```shell
************* Module example.test
example/test.py:19:8: C0103: Attribute name "thisIsABadField" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

I believe that if I disable pylint warnings for `invalid-name` where I declare a class level attribute, that should suppress all errors where I use that attribute. In this case I am provided an error for accessing the attribute in my `do_something()`, however I am not provided an error in my `do_another()`, which seems oddly strange because if the error is already getting shown for one of these it probably should have been for both. Regardless, I would expect no errors to be shown if I disable the invalid name error where I declared the attribute.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.6
Python 3.10.12 (main, Jun  7 2023, 12:45:35) [GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by mbyrnepr2** [comment]
Thanks for this issue @serbboy23!
You're using the `disable` within the scope of a class and that works fine so long as you omit line 13 in your example:
```python
# pylint: enable=invalid-name
```
This is consistent with the [function scope example in the docs](https://pylint.readthedocs.io/en/latest/user_guide/messages/message_control.html#block-disables).

There are further examples in that link which demonstrate where using the `enable` pragma within a block can be used to customise the behaviour; but in your example, that line is not required; it is however, as demonstrated, having the effect of a false positive in one of the places within the scope of the class where the name is referenced - that shouldn't happen.

**Comment by serbboy23** [comment]
> Thanks for this issue @serbboy23! You're using the `disable` within the scope of a class and that works fine so long as you omit line 13 in your example:
> 
> ```python
> # pylint: enable=invalid-name
> ```
> 
> This is consistent with the [function scope example in the docs](https://pylint.readthedocs.io/en/latest/user_guide/messages/message_control.html#block-disables).
> 
> There are further examples in that link which demonstrate where using the `enable` pragma within a block can be used to customise the behaviour; but in your example, that line is not required.

I see now that I could have used `# pylint: disable-next=invalid-name` to achieve just altering the next line. In my current example, the reason I have the disable/enable is because there are only a small subset of members that I want to enable non snake-casing for.

In the case of having a class that inherits from another, false positives are also present:

```python

"""
This module shows a bug in pylint.
"""


class Base:
    """
    This is where the invalid-name field is defined.
    """

    # pylint: disable=invalid-name
    thisIsABadField: str

    def __init__(self, greeting: str):
        print(greeting)


class Derived(Base):
    """
    This is the class that inherits from Base and has an invalid-field.
    """

    def __init__(self):
        super().__init__("What's Good")

    def do_something(self):
        self.thisIsABadField = "Hello"

```

Output:
```bash
************* Module test
test.py:27:8: C0103: Attribute name "thisIsABadField" doesn't conform to snake_case naming style (invalid-name)
```

**Comment by mbyrnepr2** [comment]
It sounds like your use-case calls for the [good-names](https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#good-names) option where you can specify a list of names which are to be excluded from the snake_case requirement.

Aside from that I've noticed that the false positive also appears when **no** pragmas are used in the example.
